lib/react/jsx.rb

```diff
@@ -14,11 +14,22 @@ module JSX
     # to provide your own transformer. It must implement:
     # - #initialize(options)
     # - #transform(code) => new code
+    #
+    # If you want to configure per-file transformation options with a proc, you may
+    # accept an optional second parameter
+    # - #transform(code, local_options = {}) => new code
     self.transformer_class = DEFAULT_TRANSFORMER
 
-    def self.transform(code)
+    def self.transform(code, input = {})
       self.transformer ||= transformer_class.new(transform_options)
-      self.transformer.transform(code)
+
+      if self.transformer.method(:transform).arity == 1
```

Comment:
How about changing `JSXTransformer#transform` to this new signature instead of checking method arity? I think the resulting code will be a bit simpler. It's _technically_ a backwards-incompatibility but I don't know of any other transformers out there! (Just `NullTransformer` in the tests :P). And the "upgrade" would be quite easy if you needed it.